src/main/java/ch/njol/skript/classes/data/SkriptClasses.java

```diff
@@ -853,6 +856,74 @@ public String getVariableNamePattern() {
 				.since("2.5")
 				.serializer(new YggdrasilSerializer<GameruleValue>())
 		);
+		
+		if (Skript.isRunningMinecraft(1, 13)) {
+			Classes.registerClass(new ClassInfo<>(StateType.class, "blockstatetype")
+					.user("block ?state ?types?")
+					.name("Block State Type")
+					.description("Represents the different states of a block's BlockData.")
+					.usage(BlockDataUtils.getStateTypeNames())
+					.examples("set lit state of target block to false",
+							"set waterlogged state of target block to true")
+					.since("INSERT VERSION")
+					.requiredPlugins("Minecraft 1.13+")
+					.parser(new Parser<StateType>() {
+						@Nullable
+						@Override
+						public StateType parse(String s, ParseContext context) {
+							return BlockDataUtils.getByName(s.replace("_", " "));
+						}
+						
+						@Override
+						public String toString(StateType o, int flags) {
+							return o.getName();
+						}
+						
+						@Override
+						public String toVariableNameString(StateType o) {
+							return "blockstatetype: " +  o.toString();
+						}
+						
+						@Override
+						public String getVariableNamePattern() {
+							return "blockstatetype:+.";
+						}
+					})
+					.serializer(new YggdrasilSerializer<>()));
+			
+			Classes.registerClass(new ClassInfo<>(StateValues.class, "blockstatevalue")
+					.user("block ?state ?values?")
+					.name("Block State Value")
+					.description("Represents different block state values which currently do not match a Skript type.")
+					.usage(StateValues.getNames())
+					.examples("set axis state of target block to x",
+							"set half state of target block to upper",
+							"set orientation state of block at player to up north")
+					.since("INSERT VERSION")
+					.requiredPlugins("Minecraft 1.13+")
+					.parser(new Parser<StateValues>() {
+						@Nullable
+						@Override
+						public StateValues parse(String s, ParseContext context) {
+							return StateValues.getByName(s.replace("_", " "));
+						}
+						
+						@Override
+						public String toString(StateValues o, int flags) {
+							return o.toString();
+						}
+						
+						@Override
+						public String toVariableNameString(StateValues o) {
+							return o.toString();
+						}
+						
+						@Override
+						public String getVariableNamePattern() {
+							return ".*";
```

Comment:
Normally don't SkriptClasses just return ``\\S+`` for this?

Reply:
Right, this was just more so for resting. Its not done yet.